idunn/geocoder/nlu_client.py

```diff
@@ -119,44 +113,43 @@ def build_intention_category_place(self, tags_list, lang):
                 description={"query": cat_query, "place": place},
             )
 
-    def build_intention_category(self, tags_list, lang):
+    async def build_intention_category(self, tags_list, lang):
         category_tag = next(t for t in tags_list if t.get("tag") == "cat")
         cat_query = category_tag["phrase"]
-        category_name = self.classify_category(cat_query)
+        category_name = await self.classify_category(cat_query)
         if category_name:
             return Intention(
                 filter={"category": category_name}, description={"category": category_name},
             )
         return None
 
-    def get_intentions(self, text, lang):
+    async def get_intentions(self, text, lang):
         tokenizer_url = settings["NLU_TOKENIZER_URL"]
         # this settings is an immutable string required as a parameter for the NLU API
         params = {"text": text, "lang": lang or settings["DEFAULT_LANGUAGE"], "domain": "poi"}
 
         try:
-            response_nlu = self.session.post(tokenizer_url, json=params, timeout=self.timeout)
+            response_nlu = await self.client.post(tokenizer_url, json=params)
             response_nlu.raise_for_status()
         except Exception:
             logger.error("Request to NLU tokenizer failed", exc_info=True)
             return []
-        else:
-            intentions = []
-            tags_list = response_nlu.json()["NLU"]
-            tags_list = [t for t in tags_list if t["tag"] != "O"]
-            tags_count = Counter(t["tag"] for t in tags_list)
-            if tags_count == {"city": 1, "cat": 1}:
-                # 1 category + 1 place
-                intention = self.build_intention_category_place(tags_list, lang=lang)
-                if intention is not None:
-                    intentions.append(intention)
-            elif tags_count == {"cat": 1}:
-                # 1 category
-                intention = self.build_intention_category(tags_list, lang=lang)
-                if intention is not None:
-                    intentions.append(intention)
-
-            return intentions
+
+        intentions = []
+        tags_list = response_nlu.json()["NLU"]
+        tags_list = [t for t in tags_list if t["tag"] != "O"]
```

Comment:
Just realized: can't you do this directly when getting `tags_list` on the previous line?

Reply:
updated